src/main/scala/fm/xml/XmlReader.scala

```diff
@@ -64,143 +101,186 @@ object XmlReader {
+    */
   private class DefaultNamespaceStreamReaderDelegate(self: XMLStreamReader2, defaultNamespaceURI: String) extends StreamReader2Delegate(self) {
     Predef.require(defaultNamespaceURI.isNotBlank, "Expected defaultNamespace to be not blank")
-    
+
     override def getNamespaceURI(): String = {
       val ns: String = super.getNamespaceURI()
       if (null == ns) defaultNamespaceURI else ns
     }
-    
+
   }
-  
+
   /**
-   * Similar to DefaultNamespaceStreamReaderDelegate but just completely overrides the namespace.  Used when people set the namespace 
-   * to something like "www.aftermarket.org" when it should be "http://www.aftermarket.org"
-   */
+    * Similar to DefaultNamespaceStreamReaderDelegate but just completely overrides the namespace.  Used when people set the namespace
+    * to something like "www.aftermarket.org" when it should be "http://www.aftermarket.org"
+    */
   private class OverrideDefaultNamespaceStreamReaderDelegate(self: XMLStreamReader2, overrideDefaultNamespaceURI: String) extends StreamReader2Delegate(self) {
     Predef.require(overrideDefaultNamespaceURI.isNotBlank, "Expected overrideDefaultNamespaceURI to be not blank")
-    
+
     override def getNamespaceURI(): String = {
       // If the prefix is blank (which I thinks means we are using the default namespace) use the override
       if (getPrefix().isBlank) overrideDefaultNamespaceURI else super.getNamespaceURI()
     }
   }
 }
 
+
+object XmlReaderPath {
+  // Apply helper for reading a non-mapped value
+  def apply[XmlValue: ClassTag](itemPath: String): XmlReaderPath[XmlValue, XmlValue] = XmlReaderPath[XmlValue, XmlValue](itemPath, { v: XmlValue => v })
+}
+
 /**
- * defaultNamespaceURI - If no namespace is set, use this as a default
- * overrideDefaultNamespaceURI - Override the namespace and just use this instead
- */
-class XmlReader[T: ClassTag](rootName: String, itemPath: String, defaultNamespaceURI: String, overrideDefaultNamespaceURI: String, protected val resource: Resource[Reader]) extends ResourceLazySeq[T, Reader] with Logging {
-  
-  if (defaultNamespaceURI.isNotBlank) require(overrideDefaultNamespaceURI.isBlank, "Can't set both defaultNamespaceURI and overrideDefaultNamespaceURI")
-  if (overrideDefaultNamespaceURI.isNotBlank) require(defaultNamespaceURI.isBlank, "Can't set both defaultNamespaceURI and overrideDefaultNamespaceURI")
-  
-  private[this] val itemClass: Class[T] = classTag[T].runtimeClass.asInstanceOf[Class[T]]
+  *
+  * @param itemPath The XPath-like path to the element we are interested in.  (e.g. "part", "items/part", etc)
+  * @param toMappedValue Map the XmlValue into another return type, this should match the return type in MultiXmlReader
+  * @tparam XmlValue The unmarshalled element class type
+  * @tparam MappedXmlValue A mapped XmlValue type
+  */
+final case class XmlReaderPath[
+XmlValue: ClassTag,
+MappedXmlValue: ClassTag,
```

Comment:
Why is `: ClassTag` being used here?

Reply:
What's the purpose of this `MappedXmlValue`?

Reply:
I don't understand.  What boilerplate?  What does this mean?

> each of the mapping values need to be done for each individual parameter - NOT the expected type that is being returned - e.g. /items/item !== //item if that's the path you specify

Reply:
Consider two paths `/Item/DigitalAsset` and `/app/DigitalAsset` both could be the same return type, but have two separate "return values" because you want to know it's from the /item/* vs the /app/* path.  This happens in the `MarketCopy` vs the PIES `Item`

Reply:
Maybe I need a new term too, but the explicit scala return type needs to be different than the "reference value type"

Reply:
I removed `ClassTag` from the `MappedXmlValue` as well as the return typing for the underlying `LazySeqReader[T, Reader]`  and `XmlReader[T]`/etc.